### PR TITLE
Add Business id field for organizations

### PR DESCRIPTION
**Pull request by quaxsze**
Close https://github.com/etalab/data.gouv.fr/issues/1120

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 - Add optionnal harvest validation form [#2864](https://github.com/opendatateam/udata/pull/2864)
 - Fix dataset list default sorting [#2867](https://github.com/opendatateam/udata/pull/2867)
 - Update API doc link [#2866](https://github.com/opendatateam/udata/pull/2866)
+- Add `establishment_number_id` metadata for organizations [#2871](https://github.com/opendatateam/udata/pull/2871)
```
Comment (geoffreyaldebert) [inline CHANGELOG.md:12]: not fan of `establishment` keyword. Can we use `company` instead ?

File: udata/core/organization/forms.py [hunk 2]
```diff
@@ -27,6 +49,9 @@ class OrganizationForm(ModelForm):
     url = fields.URLField(
         _('Website'), description=_('The organization website URL'))
     logo = fields.ImageField(_('Logo'), sizes=LOGO_SIZES)
+    business_number_id = fields.StringField(_('Business id'),
+                                                 [validators.Length(max=ORG_BID_SIZE_LIMIT), org_bid_check],
```
Comment (maudetes) [inline udata/core/organization/forms.py:53]: Why not set `max` and `min` here?

File: udata/core/organization/models.py [hunk 3]
```diff
@@ -104,6 +107,7 @@ class Organization(WithMetrics, BadgeMixin, db.Datetimed, db.Document):
     image_url = db.StringField()
     logo = db.ImageField(fs=avatars, basename=default_image_basename,
                          max_size=LOGO_MAX_SIZE, thumbnails=LOGO_SIZES)
+    business_number_id = db.StringField(max_length=128)
```
Comment (maudetes) [inline udata/core/organization/models.py:110]: Why not ORG_BID_SIZE_LIMIT instead of 128?

Comment (maudetes) [reply udata/core/organization/models.py:110]: Should we use a SIRETField for some validation at the model level?

**Review by maudetes — COMMENTED** [review]
Could you please add some tests?
In particular on successful and failing SIRET and La Poste edge case :)

File: udata/core/organization/forms.py [hunk 4]
```diff
@@ -27,6 +45,9 @@ class OrganizationForm(ModelForm):
     url = fields.URLField(
         _('Website'), description=_('The organization website URL'))
     logo = fields.ImageField(_('Logo'), sizes=LOGO_SIZES)
+    business_number_id = fields.StringField(_('Business id'),
+                                                 [validators.Length(min=ORG_BID_SIZE_LIMIT, max=ORG_BID_SIZE_LIMIT), org_bid_check],
```
Comment (maudetes) [inline udata/core/organization/forms.py:49]: Seems like I can't unset a business_number_id, I get a `Le champ doit contenir exactement 14 caractères.` error

File: udata/core/organization/forms.py [hunk 5]
```diff
@@ -15,6 +15,24 @@
...
+def org_bid_check(form, field):
+    if field.data:
+        # EID checks are country dependant. Following one is suitable for France.
+        if ORG_BID_FORMAT == 'siret':
+            siret_number = str(field.data)
+            # Checksum verification on only the SIREN part, the 9 first digits.
+            try:
+                chiffres = [int(chiffre) for chiffre in siret_number[:9]]
+                chiffres[1::2] = [chiffre * 2 for chiffre in chiffres[1::2]]
+                chiffres = [chiffre - 9 if chiffre > 9 else chiffre for chiffre in chiffres]
```
Comment (maudetes) [inline udata/core/organization/forms.py:27]: why `chiffre - 9` here?

File: udata/core/organization/forms.py [hunk 6]
```diff
@@ -15,6 +15,24 @@
 )
 
 
+def org_bid_check(form, field):
+    if field.data:
+        # EID checks are country dependant. Following one is suitable for France.
+        if ORG_BID_FORMAT == 'siret':
+            siret_number = str(field.data)
+            # Checksum verification on only the SIREN part, the 9 first digits.
```
Comment (maudetes) [inline udata/core/organization/forms.py:23]: I'm confused about this, we should be able to check for the entire SIRET, with La Poste as the only exception right?

**Review by maudetes — APPROVED** [review]
Nice, thank you! :rocket:
